test/src/test/java/hudson/cli/ConsoleCommandTest.java

```diff
@@ -229,14 +244,18 @@
     @Test public void consoleShouldSuccessWithLastNLines() throws Exception {
 
         FreeStyleProject project = j.createFreeStyleProject("aProject");
-        project.getBuildersList().add(new Shell("echo 1\necho 2\necho 3\necho 4\necho 5"));
+        if(Functions.isWindows()) {
+            project.getBuildersList().add(new BatchFile("echo 1\r\necho 2\r\necho 3\r\necho 4\r\necho 5"));
+        } else {
+            project.getBuildersList().add(new Shell("echo 1\necho 2\necho 3\necho 4\necho 5"));
+        }
         project.scheduleBuild2(0).get();
         assertThat(project.getBuildByNumber(1).getLog(), containsString("echo 1"));
         assertThat(project.getBuildByNumber(1).getLog(), containsString("echo 5"));
 
         final CLICommandInvoker.Result result = command
                 .authorizedTo(Jenkins.READ, Job.READ, Item.BUILD)
-                .invokeWithArgs("aProject", "1", "-n", "4");
+                .invokeWithArgs("aProject", "1", "-n", Functions.isWindows() ? "5" : "4");
```

Comment:
Because batch files output things differently than a shell script, so the information needed was in another line up.